optaplanner-examples/src/main/java/org/optaplanner/examples/vehiclerouting/persistence/VehicleRoutingSolutionFileIO.java

```diff
@@ -0,0 +1,112 @@
+package org.optaplanner.examples.vehiclerouting.persistence;
+
+import java.io.File;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import org.optaplanner.examples.vehiclerouting.domain.VehicleRoutingSolution;
+import org.optaplanner.examples.vehiclerouting.domain.location.DistanceType;
+import org.optaplanner.examples.vehiclerouting.domain.location.Location;
+import org.optaplanner.examples.vehiclerouting.domain.location.RoadLocation;
+import org.optaplanner.examples.vehiclerouting.domain.location.segmented.HubSegmentLocation;
+import org.optaplanner.examples.vehiclerouting.domain.location.segmented.RoadSegmentLocation;
+import org.optaplanner.persistence.jackson.impl.domain.solution.JacksonSolutionFileIO;
+
+public class VehicleRoutingSolutionFileIO extends JacksonSolutionFileIO<VehicleRoutingSolution> {
```

Comment:
That is a lot of code for an example. :-( I starting to wonder if this Jackson thing is good for the examples and therefore the users.

Reply:
I see your point, but there is a couple of things to realize:
- we don't have a better option than JSON, do we?
- it's a lot of code because there are multiple maps to update in the case of the segmented location

Reply:
I find it strange that XStream didn't need any of this. How could that be?
Anyway, that's mostly just a philosophical question. Unless we can reduce the amount of code by introducing some more reuse, we don't have other options anyway.

Reply:
XStream uses reflection heavily; we even have some not-so-nice substitutions [1] in place to make it work on GraalVM.

[1] https://github.com/kiegroup/optaplanner/blob/main/optaplanner-persistence/optaplanner-persistence-xstream/src/main/java/org/optaplanner/persistence/xstream/runtime/graal/XStreamSubstitutions.java